inference/enterprise/workflows/execution_engine/introspection/blocks_loader.py

```diff
@@ -0,0 +1,254 @@
+            public_message=f"It is not possible to load workflow plugin `{plugin_name}`. "
+            f"Make sure the library providing custom step is correctly installed in Python environment.",
+            context="workflow_compilation | blocks_loading",
+            inner_error=e,
+        ) from e
+
+
+def _load_initializers_from_plugin(
+    plugin_name: str,
+) -> Dict[str, Callable[[None], Any]]:
+    module = importlib.import_module(plugin_name)
+    registered_initializers = getattr(module, "REGISTERED_INITIALIZERS", {})
+    return {
+        f"{plugin_name}.{parameter_name}": initializer
+        for parameter_name, initializer in registered_initializers.items()
+    }
+
+
+def _validate_loaded_blocks_names_uniqueness(blocks: List[BlockDescription]) -> None:
+    block_names_lookup = {}
+    for block in blocks:
+        if block.human_friendly_block_name in block_names_lookup:
+            clashing_block = block_names_lookup[block.human_friendly_block_name]
+            raise PluginLoadingError(
+                public_message=f"Block defined in {block.block_source} plugin with fully qualified class "
+                f"name {block.fully_qualified_block_class_name} clashes in terms of "
+                f"the human friendly name (value={block.human_friendly_block_name}) with other "
+                f"block - defined in {clashing_block.block_source} with fully qualified class name: "
+                f"{clashing_block.fully_qualified_block_class_name}.",
+                context="workflow_compilation | blocks_loading",
+            )
+        block_names_lookup[block.human_friendly_block_name] = block
+    return None
+
+
+def _validate_loaded_blocks_manifest_type_identifiers(
+    blocks: List[BlockDescription],
+) -> None:
+    types_already_defined = {}
+    for block in blocks:
+        all_types = [
+            block.manifest_type_identifier
+        ] + block.manifest_type_identifier_aliases
+        for type_name in all_types:
+            if type_name in types_already_defined:
+                clashing_block = types_already_defined[type_name]
+                raise PluginLoadingError(
+                    public_message=f"Block defined in {block.block_source} plugin with fully qualified class "
+                    f"name {block.fully_qualified_block_class_name} clashes in terms of "
+                    f"the manifest type identifier (or its alias): {type_name} - defined in "
+                    f"{clashing_block.block_source} with fully qualified class name: "
+                    f"{clashing_block.fully_qualified_block_class_name}.",
+                    context="workflow_compilation | blocks_loading",
+                )
+            types_already_defined[type_name] = block
+    return None
+
+
+def _validate_used_kinds_uniqueness(declared_kinds: List[Kind]) -> None:
+    kinds_names_counter = Counter(k.name for k in declared_kinds)
+    non_unique_kinds = [k for k, v in kinds_names_counter.items() if v > 1]
```

Comment:
could do this same way as you have done a few lines above
```python
    kinds_names = [k.name for k in declared_kinds]
    if len(declared_kinds) != len(set(declared_kinds)):
        raise
```

Reply:
yes, but my point was to deliver problematic elements in error message to provide more context